apache/cordova-android
# fix: gradle build tools config

### Platforms affected

android

### Motivation and Context



Fixes a regression outlined in https://github.com/apache/cordova-android/issues/1290

Closes https://github.com/apache/cordova-android/issues/1290

While cordova-android always used the latest build tools version by default, through gradle parameters users could pin their build tools version to a specific version. In 10.x this behaviour was changed so that the gradle parameter value was used as hte minimum version instead of the desired build tools version.

### Description


I added a `MIN_BUILD_TOOLS_VERSION` as our declared minimum build tools version. The previously declared `BUILD_TOOLS_VERSION` now is unset by default, but will become either the latest build tools version that is available or the declared build tools version configured by `config.xml` preference or the gradle args CLI parameter.

`LATEST_INSTALLED_BUILD_TOOLS` config variable was completely removed in favour of `BUILD_TOOLS_VERSION`

The above changes were made in effort to not introduce anything breaking.

Requires https://github.com/apache/cordova-android/pull/1294 to fix CI testing.

### Testing


Ran npm test successfully.

Manually testing the following cases:

- Default configuration, latest build tools version `29` errors out as expected (min 30.0.3)
- Default configuration, latest build tools version `30.0.3` runs successfully.
- Default configuration, latest build tools version `31` runs successfully as expected (uses 30.0.3)
- `android-buildtoolsversion` preference 29, errors out as expected (min 30.0.3)
- `android-buildtoolsversion` preference 30.0.3 runs successfully
- `--gradleArg=-PcdvBuildToolsVersion=29.0.3` errors out as expected (min 30.0.3)
- `--gradleArg=-PcdvBuildToolsVersion=30.0.3` runs successfully


### Checklist
